tools/filterdiff.py

```diff
@@ -0,0 +1,161 @@
+# /usr/bin/env python3
+import sys
+import os
+import argparse
+import tempfile
+import json
+import subprocess
+from typing import Dict, Optional, Set, List
+
+
+class FilterImageInfo:
+    """
+    Filter specific fields in the image-info that can't be compared
+    together.
+    """
+
+    def __init__(self, imi: Optional[Dict], to_remove_tmpfiles_d: Set) -> None:
+        self.imi: Dict = imi if imi else {}
+        self.to_remove_tmpfiles_d = to_remove_tmpfiles_d
+
+    def _tmpfiles_d(self):
+        """
+        Remove certain files from the tpmfiles_d that have a tendency of having
+        a non fixed content.
+        """
+        tmpfilesd = self.imi.get("tmpfiles.d")
+        if tmpfilesd:
+            for _, v in tmpfilesd.items():
+                for trm in self.to_remove_tmpfiles_d:
+                    if trm in v:
+                        del v[trm]
+
+    def _lvm2(self):
+        """
+        LVM2 partitions have a UUID that is not fixed. Replace the value
+        upon comparison time.
+        """
+        partitions = self.imi.get("partitions")
+        if partitions:
+            for partition in partitions:
+                if partition.get("fstype") == "LVM2_member":
+                    partition["uuid"] = "2022-07-01-fixed-uuid"
+
+    def _iso(self):
+        """
+        For isos, the partition UUID is the date of the build. Replace
+        that with a fixed one for the comparison.
+        """
+        if ("image-format" in self.imi and "type" in self.imi["image-format"] and
+                self.imi["image-format"]["type"] == "raw"):
+            if "partitions" in self.imi:
+                for partition in self.imi["partitions"]:
+                    if "fstype" in partition:
+                        if partition["fstype"] == "iso9660":
+                            partition["uuid"] = "2022-07-01-fixed-uuid"
+
+    def apply(self) -> Dict:
+        self._iso()
+        self._lvm2()
+        self._tmpfiles_d()
+        return self.imi
+
+
+class ImageInfo:
+
+    imi: Dict
+    unwanted_tmpfiles_d: Set
+
+    def __init__(self, imi: Dict, unwanted_tmpfiles_d: Set, do_filter=True) -> None:
+        self.unwanted_tmpfiles_d = unwanted_tmpfiles_d
+        self.imi = imi
+        self.do_filter = do_filter
+
+    @classmethod
+    def from_file(cls, path):
+        with open(path, "r") as f:
+            c = json.load(f)
+        # Supports raw image-info or DB entry
+        if "image-info" in c:
+            return ImageInfo(
+                c["image-info"],
+                set(c.get("unwanted_tmpfiles_d", []))
+            )
+        return ImageInfo(
+            c,
+            set()
+        )
+
+    def union_unwanted_tmpfiles_d(self, set_lst: List[Set[str]]) -> Set[str]:
```

Comment:
nitpick: I don't think that saving the one letter `i` from `list` to make it `lst` is worth the decreased readability. You could have called it just `sets`.